server/users.ts

```diff
@@ -356,6 +358,9 @@ export class User extends Chat.MessageContext {
 	lastDisconnected: number;
 	lastConnected: number;
 	foodfight?: {generatedTeam: string[], dish: string, ingredients: string[], timestamp: number};
+	// We want this to be null on creation so that the database can be queried only once per user object.
+	friends: Set<string> | null;
+	friendRequests: {sent: Set<string>, received: Set<string>} | null;
```

Comment:
Can you elaborate on why these have to be null? The code is littered with `.friends?`/etc

Reply:
friendRequests is null because it's deleted if you're disabling friendRequests. 
friends is initialized as null as a way to slow how often the db is queried (the loginfilter returns undefined immediately if user.friends exists already( but is set after login.